spyder/plugins/completion/languageserver/client.py

```diff
@@ -327,35 +326,45 @@ def is_stdio_alive(self):
     def is_tcp_alive(self):
         """Detect if a tcp server is alive."""
         alive = True
-        if self.lsp_server is not None:
-            if self.lsp_server.poll() is not None:
+        if self.server is not None:
+            if self.server.state() == QProcess.NotRunning:
                 alive = False
```

Comment:
How about:

```python
        alive = False
        if self.server is not None:
            alive = self.server.state() != QProcess.NotRunning:
```

Why is the default value True for alive. If there is no transport, it will return True which feels weird.

Reply:
I refactored the code a little bit to improve the readability of this method.